apps/server/lib/lexical/server/project/diagnostics.ex

```diff
@@ -114,13 +59,20 @@ defmodule Lexical.Server.Project.Diagnostics do
   end
 
   @impl GenServer
-  def handle_info(file_diagnostics(uri: uri, diagnostics: diagnostics), %State{} = state) do
-    state = State.clear(state, uri)
-
+  def handle_info(
+        file_diagnostics(uri: uri, build_number: build_number, diagnostics: diagnostics),
+        %State{} = state
+      ) do
     state =
-      Enum.reduce(diagnostics, state, fn diagnostic, state ->
-        State.add(state, diagnostic)
-      end)
+      case diagnostics do
+        [] ->
+          State.clear(state, uri)
```

Comment:
Yes, now the flashing situation seems to have disappeared in editing mode. However, I don't quite understand why it can be cleared directly when empty.

Reply:
Empty means we've run the plugins and normal diagnostics and they've returned nothing.